runtime/tests/integration_tests.rs

```diff
@@ -174,17 +174,16 @@ fn test_invoke_dpe_get_profile_cmd() {
         ..cmd
     };
 
-    let resp = model
+    let mut resp = model
         .mailbox_execute(u32::from(CommandId::INVOKE_DPE), cmd.as_bytes())
         .unwrap()
         .expect("We should have received a response");
+    assert!(resp.len() <= std::mem::size_of::<InvokeDpeResp>());
+    resp.resize(std::mem::size_of::<InvokeDpeResp>(), 0);
 
-    let resp_hdr: &InvokeDpeResp = LayoutVerified::<&[u8], InvokeDpeResp>::new(resp.as_bytes())
-        .unwrap()
-        .into_ref();
+    let resp_hdr = InvokeDpeResp::read_from(resp.as_bytes()).unwrap();
```

Comment:
@korran I'm interested if you have any suggestions for a cleaner way to do this.

InvokeDpeResp has a data field with a max size, and we only want to write the bytes to the mailbox that are actually populated (so that we don't have to send/receive unnecessary null bytes over the mailbox). On the Caliptra side, this is fine.

On the caller side it would be size if the caller could cleanly use zerocopy to say "Populate the header and the first N bytes of `data` with the response buffer. For the remaining bytes, fill it with zeros." I couldn't find a nice way to do this, so I just resized the response Vec to be larger.

Reply:
You should be able to copy directly into the InvokeDpeResp struct:

```
let mut resp_hdr = InvokeDpeResp::default();
resp_hdr.as_bytes_mut()[..resp.len()].copy_from_slice(&resp)
```

You probably still want to enforce that resp.len() be an expected value, no?

Reply:
FWIW, you probably want some sort of helper function:

```
trait MailboxCommand {
  const ID: u32;
  type Request: zerocopy::AsBytes;
  type Response: zerocopy::AsBytes;
}

fn mailbox_invoke<T: MailboxCommand>(req: &T::Request) -> T::Response;
```

Reply:
Ah, ya that's much better :) Thanks!

Reply:
Done. I will follow up with a helper and reoeganize all the callers.